### ISSUE TITLE
Memory dosen't release after call ngt_close_index.

### ISSUE THREAD

**Issue by monochromegane** [yahoojapan/NGT#8]
I use C api and call open and close index function.
But memory dosen't release after call ngt_close_index.
I expect memory release on call that.

After opening the database of about 900 MB, I examined the memory usage at the time of closing by the `free` command.

The environment and programs are as follows:

```
$ uname -a
Linux localhost.localdomain 3.10.0-693.2.1.el7.x86_64 #1 SMP Wed Sep 6 20:06:13 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

$ cat /etc/redhat-release
CentOS Linux release 7.4.1708 (Core)
```

```c
#include "NGT/Capi.h"
#include <stdlib.h>
#include <stdio.h>
int main(int argc, char *argv[]){
   char str[256];

   NGTError err = ngt_create_error_object();
   NGTIndex index  = ngt_open_index("/path/to/my/database", err);

   printf("Opened\n");
   scanf("%s", str); // -> used 997M

   ngt_close_index(index);
   ngt_destroy_error_object(err);

   printf("Closed\n");
   scanf("%s", str); // -> used 966M. I expect 73M.

   return(0);
}
```

I execute `gcc -L/usr/local/lib -lstdc++ -lngt hoge.c && LD_LIBRARY_PATH=/usr/local/lib ./a.out` and `free` command.

I get output of free command:

```
# Opened
$ free -h
              total        used        free      shared  buff/cache   available
Mem:           4.7G        997M        1.8G        708K        1.9G        3.5G
Swap:          1.5G         58M        1.4G

# Closed
$ free -h
              total        used        free      shared  buff/cache   available
Mem:           4.7G        966M        1.8G        708K        1.9G        3.5G
Swap:          1.5G         58M        1.4G

# Exited program.
$ free -h
              total        used        free      shared  buff/cache   available
Mem:           4.7G         73M        2.7G        708K        1.9G        4.4G
Swap:          1.5G         58M        1.4G
```

Thank you for publishing a wonderful product.

**Comment by monochromegane** [comment]
I install NGT as the following:

```
$ wget https://github.com/yahoojapan/NGT/archive/v1.2.1.zip
$ unzip NGT-1.2.1.zip
$ cd NGT-1.2.1
$ mkdir build
$ cd build
$ cmake ..
$ make
$ make install
```

**Comment by masajiro** [comment]
Thank you for your comment. We will check the issue you described ASAP.

**Comment by masajiro** [comment]
We have checked the problem. Even though we reproduced the problem you mentioned, a memory leak detecting tool could not find any memory leak. Moreover we could reproduce the same problem by using a simple bug-free program without NGT. Thus, we have concluded that some memory blocks that are freed by a program cannot be actually released for the virtual memory space, because the memory blocks might be fragmented among active memory blocks.
